ultralytics/models/yolo/segment/val.py

```diff
@@ -212,17 +211,78 @@ def pred_to_json(self, predn: dict[str, torch.Tensor], pbatch: dict[str, Any]) -
             predn (dict[str, torch.Tensor]): Predictions containing bboxes, masks, confidence scores, and classes.
             pbatch (dict[str, Any]): Batch dictionary containing 'imgsz', 'ori_shape', 'ratio_pad', and 'im_file'.
         """
-        from faster_coco_eval.core.mask import encode
 
-        def single_encode(x):
-            """Encode predicted masks as RLE and append results to jdict."""
-            rle = encode(np.asarray(x[:, :, None], order="F", dtype="uint8"))[0]
-            rle["counts"] = rle["counts"].decode("utf-8")
-            return rle
-
-        pred_masks = np.transpose(predn["masks"], (2, 0, 1))
-        with ThreadPool(NUM_THREADS) as pool:
-            rles = pool.map(single_encode, pred_masks)
+        def to_string(counts: list[int]) -> str:
+            """Converts the RLE object into a compact string representation. Each count is delta-encoded and
+            variable-length encoded as a string.
+
+            Args:
+                counts (list[int]): List of RLE counts.
+            """
+            result = []
+
+            for i in range(len(counts)):
+                x = int(counts[i])
+
+                # Apply delta encoding for all counts after the second entry
```

Comment:
❗ **CRITICAL**: The RLE "delta" step should start from the third entry (i == 2). With the current `if i > 2` guard we skip subtracting on the third count, so the compressed string diverges from the COCO spec and decoding will yield wrong masks. This will quietly corrupt `iscrowd` segmentations and COCO metrics. Please change the condition to trigger for `i > 1`.

**Suggested change:**
```suggestion
                if i > 1:
```